internal/osbuild/assembler_test.go

```diff
@@ -0,0 +1,230 @@
+			args: args{
+				data: []byte(`{"name":"org.osbuild.rawfs"`),
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing name",
+			args: args{
+				data: []byte(`{"options":{"bar":null}}`),
+			},
+			wantErr: true,
+		},
+		{
+			name: "qemu assembler empty",
+			assembler: Assembler{
+				Name:    "org.osbuild.qemu",
+				Options: &QEMUAssemblerOptions{},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.qemu","options":{"format":"","filename":"","size":0,"ptuuid":"","pttype":"","partitions":null}}`),
+			},
+		},
+		{
+			name: "qemu assembler full",
+			assembler: Assembler{
+				Name: "org.osbuild.qemu",
+				Options: &QEMUAssemblerOptions{
+					Format:   "qcow2",
+					Filename: "disk.qcow2",
+					Size:     2147483648,
+					PTUUID:   "0x14fc63d2",
+					PTType:   "mbr",
+					Partitions: []QEMUPartition{QEMUPartition{
+						Start:    2048,
+						Bootable: true,
+						Filesystem: QEMUFilesystem{
+							Type:       "ext4",
+							UUID:       "76a22bf4-f153-4541-b6c7-0332c0dfaeac",
+							Label:      "root",
+							Mountpoint: "/",
+						},
+					}},
+				},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.qemu","options":{"format":"qcow2","filename":"disk.qcow2","size":2147483648,"ptuuid":"0x14fc63d2","pttype":"mbr","partitions":[{"start":2048,"bootable":true,"filesystem":{"type":"ext4","uuid":"76a22bf4-f153-4541-b6c7-0332c0dfaeac","label":"root","mountpoint":"/"}}]}}`),
+			},
+		},
+		{
+			name: "tar assembler empty",
+			assembler: Assembler{
+				Name:    "org.osbuild.tar",
+				Options: &TarAssemblerOptions{},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.tar","options":{"filename":"","size":0}}`),
+			},
+		},
+		{
+			name: "tar assembler full",
+			assembler: Assembler{
+				Name: "org.osbuild.tar",
+				Options: &TarAssemblerOptions{
+					Filename:    "root.tar.xz",
+					Size:        0,
+					Compression: "xz",
+				},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.tar","options":{"filename":"root.tar.xz","size":0,"compression":"xz"}}`),
+			},
+		},
+		{
+			name: "rawfs assembler empty",
+			assembler: Assembler{
+				Name:    "org.osbuild.rawfs",
+				Options: &RawFSAssemblerOptions{},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.rawfs","options":{"filename":"","root_fs_uuid":"00000000-0000-0000-0000-000000000000","size":0}}`),
+			},
+		},
+		{
+			name: "rawfs assembler full",
+			assembler: Assembler{
+				Name: "org.osbuild.rawfs",
+				Options: &RawFSAssemblerOptions{
+					Filename:           "filesystem.img",
+					RootFilesystemUUID: uuid.MustParse("76a22bf4-f153-4541-b6c7-0332c0dfaeac"),
+					Size:               2147483648,
+				},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.rawfs","options":{"filename":"filesystem.img","root_fs_uuid":"76a22bf4-f153-4541-b6c7-0332c0dfaeac","size":2147483648}}`),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
```

Comment:
Note: you don't really have to create a new object here. The alternative syntax will be `assert.Nilf(t, ...)` - i.e. passing `t` as the first argument for every assert function.

I don't really know if this is only a syntax difference or there could be more implications (other than consuming a few bytes of memory)

Reply:
The two ways should be equal (https://godoc.org/github.com/stretchr/testify/assert) and that one I used can slightly improve readability, IMHO. But of course I don't insist on using it.

Reply:
Yes, both are equivalent. However you are creating a new object inside a loop and that new object always has the same arguments, namely `t`.

It would be different if that was `assert.New(tt)` - you may say that assert errors will get better annotation that points to each particular test data member in the `tests` array. However the code in question is doing the same thing on every iteration. 

If you want to keep the current syntax you can either change the arguments to New() (although I don't think there will be any practical difference) or move this line above the for loop (which will gain a miscule speed improvement).

Reply:
Ah, of course. The `assert.New` inside of the loop doesn't really make sense (and it even wasn't an intention). Fixed.

Reply:
I'm not seeing the difference.

Reply:
The difference is that I moved the instantiation of the assert object outside of the loop, so that now it's instantiated only once, before looping through all the scenarios.

Reply:
Line 112 contradicts that.

Reply:
OMG, sorry for my blindness. Fixed.